### ISSUE TITLE
editorconfigCheck fails due to incompatible version of antlr4-runtime

### ISSUE THREAD

**Issue by edwinvanderham** [ec4j/editorconfig-gradle-plugin#7]
gradlew editorConfigCheck

..

> Task :app:editorconfigCheck FAILED
ANTLR Tool version 4.7 used for code generation does not match the current runtime version 4.5.3

**Comment by ppalaga** [comment]
Could you please add steps to reproduce?

**Comment by edwinvanderham** [comment]
I have an android application with the following root build.gradle:

```
buildscript {
  repositories {
    google()
    jcenter()
  }
  dependencies {
    classpath 'com.android.tools.build:gradle:3.1.4'
 }
}

plugins {
  id 'org.ec4j.editorconfig' version '0.0.2'
}
```

The app build.gradle is nothing special:
```
apply plugin: 'com.android.application'

android {
  compileSdkVersion 26
  defaultConfig {
    applicationId "some.app.id"
    minSdkVersion 19
    targetSdkVersion 26
    versionCode 1
    versionName "1.0"
    testInstrumentationRunner "android.support.test.runner.AndroidJUnitRunner"
  }
  buildTypes {
    release {
      minifyEnabled false
      proguardFiles getDefaultProguardFile('proguard-android.txt'), 'proguard-rules.pro'
    }
  }
}

configurations.all {
  // Don't cache changing modules at all.
  resolutionStrategy.cacheChangingModulesFor 0, 'seconds'
}

dependencies {
  implementation fileTree(dir: 'libs', include: ['*.jar'])
  implementation 'com.android.support:appcompat-v7:26.1.0'
}
```
when I run the command 'gradlew editorConfigCheck' I get the following output:

```
FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':editorconfigCheck'.
> Could not initialize class org.ec4j.maven.linters.xml.XmlParser
```

**Comment by edwinvanderham** [comment]
Here's a simple reproduction scenario:
clone a google sample project
```git clone https://github.com/googlesamples/android-CardView.git```

edit Application\build.gradle:
```
buildscript {
    repositories {
        jcenter()
        google()
    }

    dependencies {
        classpath 'com.android.tools.build:gradle:3.0.1'
    }
}

plugins {
  id 'org.ec4j.editorconfig' version '0.0.2'
}

apply plugin: 'com.android.application'
...
```
Put .editorconfig file in root dir:
```
[*]
charset=utf-8
end_of_line=lf
insert_final_newline=false
indent_style=space
indent_size=2

[{*.sht,*.html,*.shtm,*.shtml,*.htm}]
indent_style=space
indent_size=4

[{*.mod,*.dtd,*.ent,*.elt}]
indent_style=space
indent_size=4

[{*.jhm,*.xslt,*.xul,*.rng,*.xsl,*.xsd,*.ant,*.tld,*.fxml,*.jrxml,*.xml,*.jnlp,*.wsdl}]
indent_style=space
indent_size=4

[{org.codehaus.groovy.runtime.ExtensionModule,*.properties}]
indent_style=space
indent_size=4
```

run gradlew editorconfigCheck:
```
> Configure project :Application
Configuration 'compile' in project ':Application' is deprecated. Use 'implementation' instead.
The CompileOptions.bootClasspath property has been deprecated and is scheduled to be removed in Gradle 5.0. Please use the CompileOptions.bootstrapClasspath property instead.

> Task :Application:editorconfigCheck FAILED
ANTLR Tool version 4.7 used for code generation does not match the current runtime version 4.5.3ANTLR Runtime version 4.7 used for parser compilation does not match the current runtime version 4.5.3

FAILURE: Build failed with an exception.
```

**Comment by ppalaga** [comment]
Thanks @edwinvanderham , the steps look good, I'll have a look in the following days.

**Comment by edwinvanderham** [comment]
Thanks a lot

**Comment by ppalaga** [comment]
I was able to reproduce the problem. It looks like the plugins is run with an incompatible version of antlr4-runtime. I am new to gradle myself and therefore I had to ask for assistance on Gradle forums: https://discuss.gradle.org/t/my-plugin-gets-an-incompatible-dependency-version-at-runtime/28543

**Comment by edwinvanderham** [comment]
Any progress on this?

**Comment by ppalaga** [comment]
Yes, the reply on Gradle forums seems to suggest a valid solution. I started fixing it several times, but always got distracted. Hopefully, I'll find enough time soon. Sorry for the delay.

**Comment by ppalaga** [comment]
I think I got the classpath isolation issue solved now, but with the `android-CardView` project, the plugin still does not work correctly. It is perhaps because there is a submodule. I am investigating.

**Comment by ppalaga** [comment]
Just released version `0.0.3` of the plugin that fixes the current issue. I tested with this branch https://github.com/ppalaga/android-CardView/commits/181029-antlr-clash of android-CardView.

Thanks again for your report.
